File: 893 - Groups of Special-Equivalent Strings.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int numSpecialEquivGroups(String[] A) {
        Set<String> s = new HashSet<>();
        for (String a : A) {
            String even = "", odd = "";
            for (int i = 0; i < a.length(); i++) {
                if (i % 2 == 0) {
                    even += a.charAt(i);
                } else {
                    odd += a.charAt(i);
                }
            }
            char[] evenArr = even.toCharArray();
            Arrays.sort(evenArr);
            char[] oddArr = odd.toCharArray();
            Arrays.sort(oddArr);
            s.add(new String(evenArr) + new String(oddArr));
        }
        return s.size();
    }
}

File: 893 - Groups of Special-Equivalent Strings.py
class Solution:
    def numSpecialEquivGroups(self, A: list[str]) -> int:
        s = set()
        for string in A:
            even = sorted(string[::2])
            odd = sorted(string[1::2])
            s.add(''.join(even + odd))
        return len(s)
